crates/driver/src/boundary/settlement.rs

```diff
@@ -245,6 +248,115 @@ impl Settlement {
             solver: self.solver,
         })
     }
+
+    pub fn with_protocol_fees(&mut self, solution: &competition::Solution) {
+        struct PricedTrade {
```

Comment:
What is this struct trying to express? The trade priced when, how and with what price in particular?